src/targets/gpu/target.cpp

```diff
@@ -105,6 +105,12 @@ std::vector<pass> target::get_passes(migraphx::context& gctx, const compile_opti
     unsupported_types.erase(shape::type_t::tuple_type);
     unsupported_types.erase(shape::type_t::bf16_type);
 
+    std::set<std::string> unsupported_bf16_ops = {};
+    if(not gpu::gfx_has_bf16_intrinsics())
+    {
+        unsupported_bf16_ops.insert("dot");
+    }
+
```

Comment:
```suggestion
    if(gpu::gfx_has_bf16_intrinsics())
    {
        unsupported_types.erase(shape::type_t::bf16_type);
    }
```
This is probably what Paul meant.  So it'll use the general eliminate_data_type pass. Can remove the other new passes below this line.

Reply:
Makes sense. Wasn't sure if we wanted to remove only the dot operation or all of BF16 for Navi21.